libs/hdf-converters/src/mappings/ChecklistMappingData.ts

```diff
@@ -0,0 +1,241 @@
+import _ from 'lodash';
+
+export type ChecklistFile = {
+  /**
+   * Unique identifier for this file. Used to encode which file is currently selected, etc.
+   *
+   * Note that in general one can assume that if a file A is loaded AFTER a file B, then
+   * A.unique_id > B.unique_id.
+   * Using this property, one might order files by order in which they were added.
+   */
+  uniqueId: string;
+  /** The filename that this file was uploaded under. */
+  filename: string;
+
+  database_id?: string;
+
+  asset: ChecklistAsset;
+  stigs: ChecklistStig[];
+  raw: Record<string, unknown>;
+};
+
+type ChecklistAsset = {
+  role: string;
+  assettype: string;
+  hostname: string;
+  hostip: string;
+  hostmac: string;
+  hostfqdn: string;
+  marking: string;
+  targetcomment: string;
+  techarea: string;
+  targetkey: string;
+  webordatabase: boolean;
+  webdbsite: string;
+  webdbinstance: string;
+};
+
+export type ChecklistStig = {
+  header: StigHeader;
+  vulns: ChecklistVuln[];
+};
+
+type StigHeader = {
+  version: string;
+  classification: string;
+  customname?: string;
+  stigid: string;
+  description: string;
+  filename: string;
+  releaseinfo?: string;
+  title: string;
+  uuid: string;
+  notice?: string;
+  source?: string;
+};
+
+type ChecklistVuln = {
+  status: string | undefined; // Look into alternate way of mapping.  Shouldn't be undefined
+  findingDetails: string;
+  comments: string;
+  severityOverride: string;
+  severityJustification: string;
+  vulnNum: string;
+  severity: string;
+  groupTitle: string;
+  ruleId: string;
+  ruleVersion: string;
+  ruleTitle: string;
+  vulnDiscuss: string;
+  iaControls: string;
+  checkContent: string;
+  fixText: string;
+  falsePositives: string;
+  falseNegatives: string;
+  documentable: string;
+  mitigations: string;
+  potentialImpact: string;
+  thirdPartyTools: string;
+  mitigationControl: string;
+  responsibility: string;
+  securityOverrideGuidance: string;
+  checkContentRef: string;
+  weight: string;
+  class: string;
+  stigRef: string;
+  targetKey: string;
+  stigUuid: string;
+  legacyId: string;
+  cciRef: string;
+};
+
+/**
+ * Gets attribute data
+ * @param stigdata - array of raw stig data json
+ * @param tag - attribute name
+ * @returns attribute data
+ */
+function getAttributeData(stigdata: unknown[], tag: string): string {
+  const results = stigdata.filter((attribute: unknown) => {
+    return _.get(attribute, 'vulnattribute') === tag;
+  });
+  if (results.length === 1) {
+    return _.get(results[0], 'attributedata') as unknown as string;
+  } else {
```

Comment:
There is nothing to prevent multiple <VULN_ATTRIBUTE> values being the same. In some cases CCI_REF will be multiple. This allows for 1+ attributes to be combined into one JSON key. The function was updated to return only the .map callback to .join each instance of the attribute name.